br/pkg/lightning/importer/chunk_process.go

```diff
@@ -256,7 +256,8 @@ func (cr *chunkProcessor) encodeLoop(
 		canDeliver := false
 		kvPacket := make([]deliveredKVs, 0, maxKvPairsCnt)
 		curOffset := offset
-		var newOffset, rowID, realOffset int64
+		var newOffset, rowID, newRealOffset int64
+		var realOffset int64 = -1
```

Comment:
what's the definition of "real" and without "real"?

Reply:
RealOffset always means the position that the current file pointer is at. Offset means the checkpoint position we have already handled. For normal files it's the offset we handled. For parquet files it's the row count we handled. For compressed files it's the uncompressed the file offset. RealOffset consumes more data than Offset.

Reply:
> RealOffset always means the position that the current file pointer is at. Offset means the checkpoint position we have already handled. For normal files it's the offset we handled. For parquet files it's the row count we handled. For compressed files it's the uncompressed the file offset. RealOffset consumes more data than Offset.

From your explanation I concluded that "position" is a logical concept, different implementation has different position. And "real" means "scanned". Is this correct?

My recommendation is
- Add the explanation of "position" to parser interface
- Rename RealPos to ScannedPos
- In future we can add another interface or type alias for "position", rather than int64, to avoid position is calcuating with bytes number

Reply:
> From your explanation I concluded that "position" is a logical concept, different implementation has different position. And "real" means "scanned". Is this correct?

Yes. Already addressed in https://github.com/pingcap/tidb/pull/42070/commits/ecb3e152216f2c542ba634c8fef731af8f336ea7.

> My recommendation is
> * Add the explanation of "position" to parser interface
> * Rename RealPos to ScannedPos

We already have comments for specific structures. I add an abstract one in this commit too.
https://github.com/pingcap/tidb/blob/ae0255cde269e40177d4036ad803e9ee6dc1eb69/br/pkg/lightning/mydump/parquet_parser.go#L320

Reply:
@lance6716 PTAL